app/views/posts/show.html.erb

```diff
@@ -14,17 +14,25 @@
 
     <p class="card-text"><%= @post.content %></p>
 
-    <p class="card-subtitle mb-2">
-      <time datetime="<%= @post.created_at.to_date %>">  
-        <%= t('posts.views.show.publish_date', publish_date: I18n.l(@post.created_at.to_datetime, format: :long)) %>
-      </time>
-    </p>
+    <% if @post.draft? %>
+      <p class="card-subtitle mb-2">
+        <time datetime="<%= @post.created_at.to_date %>">  
+          <%= date_fixer(@post) %> <%= I18n.l(@post.created_at.to_datetime, format: :long) %>
+        </time>
+      </p>
+    <% else %>
+      <p class="card-subtitle mb-2">
+        <time datetime="<%= @post.created_at.to_date %>">  
+          <%= date_fixer(@post) %> <%= I18n.l(@post.published_at ? @post.published_at.to_datetime : @post.created_at.to_datetime, format: :long) %>
```

Comment:
Aqui, no atributo `datetime`, tbm tem de fazer a mesma lógica na linha de baixo no `i18n`, pois, se na linha de baixo for `published_at.to_datetime`, os leitores de tela para pessoas cegas (accesibilidade), ouvirão uma data diferente da que está escrita,que  é independente de localização geográfica.

Reply:
Boa, é verdade